test/testcafe/framework/page-objects-v1/DeviceCodeActivatePageObject.js

```diff
@@ -48,6 +48,10 @@ export default class DeviceCodeActivatePageObject extends BasePageObject {
     return this.form.findFormFieldInput('username').visible;
   }
 
+  isPasswordFieldVisible() {
```

Comment:
looks like you need additional page object, e.g.
- `PrimaryAuthPageObject`
- `IdPDiscoveryPageObject`

Reply:
AFAIK, PageObject pattern is meant to create a Class per page.
Those method related to userName, password implies user already lands on different page (idp-discovery, primary auth etc) hence shall create separated page object.